src/components/CardGridSearch.astro

```diff
@@ -0,0 +1,255 @@
+---
+/* CSS and html structure is based on:
+ * https://github.com/CloudCannon/pagefind/blob/main/pagefind_ui/default/svelte/ui.svelte
+ * And CSS variables from Starlight from:
+ * https://github.com/withastro/starlight/blob/a0f6fa80d9be1fbe85530323a0b2523acb83a15a/packages/starlight/components/Search.astro
+ */
+
+interface Props {
+	placeholder?: string;
+	title?: string;
+}
+const { placeholder = 'Quick search a feature...', title = 'Search a feature' } = Astro.props;
+
+// TODO: Expose placeholder and title to i18n
+---
+
+<card-search>
+	<div id="search-input" class="pagefind-ui pagefind-ui--reset pagefind-ui--reset">
+		<form
+			class="pagefind-ui__form"
+			role="search"
+			aria-label="search_label"
+			action="javascript:void(0);"
+		>
+			<input
+				data-wrapper="search-feature-list"
+				class="pagefind-ui__search-input"
+				id="search"
+				aria-label="search"
+				placeholder={placeholder}
+				title={title}
+				autocapitalize="none"
+				enterkeyhint="search"
+				autocomplete="off"
+			/>
+
+			<button class="pagefind-ui__search-clear pagefind-ui__suppressed">clear</button>
+		</form>
+	</div>
+	<slot />
+</card-search>
+
+<script>
+	class CardSearch extends HTMLElement {
+		constructor() {
+			super();
+			window.addEventListener('DOMContentLoaded', () => {
+				const form = this.querySelector<HTMLFormElement>('form.pagefind-ui__form')!;
+				const searchBar = this.querySelector<HTMLInputElement>('input.pagefind-ui__search-input')!;
+				const clear = this.querySelector<HTMLButtonElement>('button.pagefind-ui__search-clear')!;
+				const inputs = Array.from(this.children).filter((el) => el.classList.contains('card-grid'));
+				const headings = Array.from(this.children).filter(
+					(el) => el instanceof HTMLHeadingElement || el instanceof HTMLParagraphElement
+				);
+				// Create "nothing found" for each heading
+				headings.forEach((heading, i) => {
+					const wrn = document.createElement('p');
+					wrn.classList.add('hidden', 'warning');
+					wrn.id = `warn-${i}`;
+					wrn.innerHTML = `
+						<style>
+							.warning {
+								color: var(--sl-color-orange-high);
+								font-size: larger;
+							}
+							.hidden {
+								display: none;
+							}
+						</style>
+						<span>Nothing found in ${heading.textContent}</span>
```

Comment:
I would style these the same as `<p>` tags, so having a color of `var(--sl-color-text)`. Could possibly make the text in italics just to set it apart a bit. 

Would also change the text to "No results found in [...]"

Reply:
preview:
![image](https://github.com/tauri-apps/tauri-docs/assets/61759797/f4884e53-b3e6-4cc5-8b5a-4133e5ea0396)